src/linker.c

```diff
@@ -239,7 +240,11 @@ static int process_dependencies(jq_state *jq, jv jq_origin, jv lib_origin, block
   block bk = *src_block;
   int nerrors = 0;
 
-  jv_array_foreach(deps, i, dep) {
+  // XXX This is a backward jv_array_foreach because bindings go in reverse
+  for (int i = jv_array_length(jv_copy(deps)); i > 0; ) {
```

Comment:
a) Maybe we ought to have a macro for this too,

b) I've long wanted to do something a wee bit safer w.r.t. lengths.  A signed length works well for a backwards foreach, but for just about everything else I'd rather use `size_t` due to it being unsigned (which has well-defined overflow semantics in C).

I never convinced @stedolan of (b) and I'm not going to wait for (a).